pennylane/devices/qubit/adjoint_jacobian.py

```diff
@@ -59,16 +93,20 @@ def adjoint_jacobian(tape: QuantumTape, state=None):
     Returns:
         array or tuple[array]: the derivative of the tape with respect to trainable parameters.
         Dimensions are ``(len(observables), len(trainable_params))``.
+
     """
     # Map wires if custom wire labels used
     tape = tape.map_to_standard_wires()
 
+    if isinstance(tape.measurements[0], qml.measurements.StateMP):
```

Comment:
do we know at this point that the tape only has 1 measurement, and has been unwrapped? wondering because I see we call `jac.flatten()` in the function, but we also raise a jax-related error in `adjoint_jvp`

Reply:
See the transform that gets applied in preprocessing.  Is there's not expectation values, it gets turned in to a circuit with a single state measurement.